### ISSUE TITLE
Iteration over get_commits() returns wrong object properties

### ISSUE THREAD

**Issue by MichaelPavlov** [PyGithub/PyGithub#773]
Here is a snippet
```
from github import Github

g = Github(token)
repo = g.get_repo('PyGithub/PyGithub')
commits = repo.get_commits()
i = 20
for commit in commits:
     print(commit.last_modified)
     i -= 1
     if i < 0:
             break
```

It prints out only first retrieved object's date. Shouldn't it print objects actual property?
```
Thu, 26 Apr 2018 06:52:47 GMT
Thu, 26 Apr 2018 06:52:47 GMT
Thu, 26 Apr 2018 06:52:47 GMT
Thu, 26 Apr 2018 06:52:47 GMT
Thu, 26 Apr 2018 06:52:47 GMT
Thu, 26 Apr 2018 06:52:47 GMT
Thu, 26 Apr 2018 06:52:47 GMT
Thu, 26 Apr 2018 06:52:47 GMT
Thu, 26 Apr 2018 06:52:47 GMT
Thu, 26 Apr 2018 06:52:47 GMT
Thu, 26 Apr 2018 06:52:47 GMT
Thu, 26 Apr 2018 06:52:47 GMT
Thu, 26 Apr 2018 06:52:47 GMT
Thu, 26 Apr 2018 06:52:47 GMT
Thu, 26 Apr 2018 06:52:47 GMT
Thu, 26 Apr 2018 06:52:47 GMT
Thu, 26 Apr 2018 06:52:47 GMT
Thu, 26 Apr 2018 06:52:47 GMT
Thu, 26 Apr 2018 06:52:47 GMT
Thu, 26 Apr 2018 06:52:47 GMT
Thu, 26 Apr 2018 06:52:47 GMT
```

**Comment by sfdye** [comment]
It seems that the `last_modified` is retrieved from the response header as part of the caching header in stead of the response body. If you inspect the header, they are indeed the same.

**Comment by MichaelPavlov** [comment]
But how can I get the value from response body in this case?

**Comment by sfdye** [comment]
Try `commit.commit.author.date` or `commit.commit.committer.date`

**Comment by sfdye** [comment]
@MichaelPavlov Does it work for you now?

**Comment by MichaelPavlov** [comment]
Yes, property `commit.commit.commiter.date` gives correct result. Thank you. But you know if I take an element like `item = get_commits()[1]' in debugger and after evaluate `item.last_modified` it gives another date. Can you explain why this happens? Like result was updated in some cache and returned after that.

**Comment by sfdye** [comment]
Probably because it's reflecting the `last-modified` attribute from the HTTP response [header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Last-Modified)? which changes with your response. If you want use the fixed date when commit was made, use any of the two properties above.

**Comment by MichaelPavlov** [comment]
But why it's necessary to call `item.update()` to get this header property correct value?

**Comment by sfdye** [comment]
I am afraid I didn't get what you meant.

**Comment by MichaelPavlov** [comment]
Maybe these prints would be more descriptive for you:

	from github import Github

	g = Github(token)
	repo = g.get_repo('PyGithub/PyGithub')
	commits = repo.get_commits()
	i = 5
	for commit in commits:
		print(commit.last_modified)
		i -= 1
		if i < 0:
			break


	Tue, 01 May 2018 01:40:08 GMT
	Tue, 01 May 2018 01:40:08 GMT
	Tue, 01 May 2018 01:40:08 GMT
	Tue, 01 May 2018 01:40:08 GMT
	Tue, 01 May 2018 01:40:08 GMT
	
	

	from github import Github

	g = Github(token)
	repo = g.get_repo('PyGithub/PyGithub')
	commits = repo.get_commits()
	i = 5
	for commit in commits:
		commit.update()
		print(commit.last_modified)
		i -= 1
		if i < 0:
			break
	
	
	Tue, 01 May 2018 01:40:08 GMT
	Mon, 30 Apr 2018 02:12:43 GMT
	Mon, 30 Apr 2018 02:10:32 GMT
	Thu, 26 Apr 2018 06:52:47 GMT
	Thu, 26 Apr 2018 06:46:44 GMT
	Thu, 26 Apr 2018 06:35:39 GMT
